packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
-        begin: _scale,
-        end: 1.0,
-      ).animate(_animationController);
-      _differenceAnimation = Tween<double>(
-        begin: _difference,
-        end: 0,
-      ).animate(_animationController);
-    });
-    if (_left > widget.threshold ||
-        _left == 0 && widget.direction == AppinioSwiperDirection.right) {
-      _swipedDirectionHorizontal = 1;
-      detectedDirection = AppinioSwiperDirection.right;
-    } else {
-      _swipedDirectionHorizontal = -1;
-      detectedDirection = AppinioSwiperDirection.left;
-    }
-    (_top <= 0) ? _swipedDirectionVertical = 1 : _swipedDirectionVertical = -1;
-    _horizontal = true;
-  }
-
-  //moves the card away to the top or bottom
-  void _swipeVertical(BuildContext context) {
-    _unSwiped = false;
-    setState(() {
-      _swipeType = 1;
-      _leftAnimation = Tween<double>(
-        begin: _left,
-        end: _left + _left,
-      ).animate(_animationController);
-      _topAnimation = Tween<double>(
-        begin: _top,
-        end: (_top == 0)
-            ? (widget.direction == AppinioSwiperDirection.bottom)
-                ? MediaQuery.of(context).size.height
-                : -MediaQuery.of(context).size.height
-            : (_top > widget.threshold)
-                ? MediaQuery.of(context).size.height
-                : -MediaQuery.of(context).size.height,
-      ).animate(_animationController);
-      _scaleAnimation = Tween<double>(
-        begin: _scale,
-        end: 1.0,
-      ).animate(_animationController);
-      _differenceAnimation = Tween<double>(
-        begin: _difference,
-        end: 0,
-      ).animate(_animationController);
-    });
-    if (_top > widget.threshold ||
-        _top == 0 && widget.direction == AppinioSwiperDirection.bottom) {
-      _swipedDirectionVertical = -1;
-      detectedDirection = AppinioSwiperDirection.bottom;
-    } else {
-      _swipedDirectionVertical = 1;
-      detectedDirection = AppinioSwiperDirection.top;
-    }
-    (_left >= 0)
-        ? _swipedDirectionHorizontal = 1
-        : _swipedDirectionHorizontal = -1;
-  }
-
-  //moves the card back to starting position
-  void _goBack(BuildContext context) {
-    setState(() {
-      _swipeType = 3;
-      _leftAnimation = Tween<double>(
-        begin: _left,
-        end: 0,
-      ).animate(_animationController);
-      _topAnimation = Tween<double>(
-        begin: _top,
-        end: 0,
-      ).animate(_animationController);
-      _scaleAnimation = Tween<double>(
-        begin: _scale,
-        end: 0.9,
-      ).animate(_animationController);
-      _differenceAnimation = Tween<double>(
-        begin: _difference,
-        end: widget.cardsSpacing,
-      ).animate(_animationController);
-    });
-    widget.onSwipeCancelled?.call();
+/// A controller used to control an [AppinioSwiper],
+///
+/// The controller notifies listeners when a swipe starts and for each tick of a
+/// swipe animation.
+class AppinioSwiperController extends ChangeNotifier {
```

Comment:
I moved the controller into this file so that it can interact with the widget state without having to make the widget state public.